File: 0305/SeoyeonHong/1697 숨바꼭질.py
from collections import deque

N, K = map(int, input().split())

if N >= K:
    print(N - K)
else: # 수빈이가 동생보다 뒤에 있을 경우
    visited = [False] * (2 * K - N)
    visited[N] = True
    q = deque([(N, 0)])
    while q:
        l, t = q.popleft()
        if l == K:
            print(t)
            break

        for i in [l*2, l+1, l-1]:
            if 0 <= i < 2 * K - N and not visited[i]:
                visited[i] = True
                q.append((i, t+1))

    

File: 0305/SoohyunKim/1697_숨바꼭질.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, K] = input[0].split(' ').map(Number);
const MAX = 100000;
const visited = Array(MAX + 1).fill(false);

function bfs(start) {
  const queue = [[start, 0]]; // (위치, 시간)
  visited[start] = true;

  while (queue.length) {
    const [curX, time] = queue.shift();

    if (curX === K) return time; // 목표 위치 도달하면 시간 반환

    for (const next of [curX + 1, curX - 1, curX * 2]) {
      if (next >= 0 && next <= MAX && !visited[next]) {
        visited[next] = true;
        queue.push([next, time + 1]);
      }
    }
  }

  return -1;
}

console.log(bfs(N));